go.mod

```diff
@@ -1,19 +1,19 @@
 module github.com/Mongey/terraform-provider-kafka
 
-go 1.22.7
+go 1.23.0
```

Comment:
sarama bumped the go version so this was required https://github.com/IBM/sarama/blob/d6ca80a198c4fbc467c67f7430182eda6881b12c/go.mod#L3